scrape/scrape.go

```diff
@@ -1027,8 +1029,15 @@ mainLoop:
 func (sl *scrapeLoop) scrapeAndReport(interval, timeout time.Duration, last time.Time, errc chan<- error) time.Time {
 	start := time.Now()
 
-	// Only record after the first scrape.
 	if !last.IsZero() {
+		// Apply some tolerance on the scrape timestamp to align the samples
+		// correctly between scrapes.
```

Comment:
This comment is wrong, this behaviour is incorrectly aligning samples as we're lying about the true timestamp. There's also no semantic requirement that samples be aligned across scrapes, indeed you should write your expressions to be resilient to jitter.

 The comment should be about why we're actually doing this.

Reply:
I have tried to add better comments.